geth/account/accounts.go

```diff
@@ -14,30 +13,37 @@ import (
 	gethcommon "github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/status-im/status-go/extkeys"
-	"github.com/status-im/status-go/geth/common"
-	"github.com/status-im/status-go/geth/rpc"
 )
 
 // errors
 var (
+	ErrInvalidAccountAddressOrKey      = errors.New("cannot parse address or key to valid account address")
 	ErrAddressToAccountMappingFailure  = errors.New("cannot retrieve a valid account for a given address")
 	ErrAccountToKeyMappingFailure      = errors.New("cannot retrieve a valid key for a given account")
 	ErrWhisperIdentityInjectionFailure = errors.New("failed to inject identity into Whisper")
 	ErrWhisperClearIdentitiesFailure   = errors.New("failed to clear whisper identities")
 	ErrNoAccountSelected               = errors.New("no account has been selected, please login")
 	ErrInvalidMasterKeyCreated         = errors.New("can not create master extended key")
+	ErrInvalidAccountManager           = errors.New("could not retrieve account manager")
+	ErrAccountKeyStoreMissing          = errors.New("account key store is not set")
 )
 
+// GethServiceProvider provides required geth services.
+type GethServiceProvider interface {
+	Account() (*accounts.Manager, error)
+	AccountKeyStore() (*keystore.KeyStore, error)
+}
+
 // Manager represents account manager interface
 type Manager struct {
-	nodeManager     common.NodeManager
-	selectedAccount *common.SelectedExtKey // account that was processed during the last call to SelectAccount()
+	geth            GethServiceProvider
+	selectedAccount *SelectedExtKey // account that was processed during the last call to SelectAccount()
 }
 
 // NewManager returns new node account manager
-func NewManager(nodeManager common.NodeManager) *Manager {
+func NewManager(geth GethServiceProvider) *Manager {
```

Comment:
Is `Manager` an implementation of `Accounter`? So `NewAccounter` would be a good name. Or is `Accounter` only a sub-interface of `Manager`?

Reply:
Yep, Manager is an implementation of `Accounter`, so, agree on changing this method to `NewAccounter`